src/test/java/com/faforever/client/test/ElideMatchers.java

```diff
@@ -0,0 +1,110 @@
+  }
+
+  public static ElideParamMatcher filterPresent() {
+    return new ElideParamMatcher(containsString("filter="));
+  }
+
+  public static ElideParamMatcher hasFilter(Condition<?> condition) {
+    return new ElideParamMatcher(containsString(String.format("filter=%s", condition.query(new RSQLVisitor()))));
+  }
+
+  public static ElideParamListMatcher<ElideNavigatorOnCollection<?>> hasSort(String sortProperty, boolean ascending) {
+    return new ElideParamListMatcher<>(contains((ascending ? "" : "-") + sortProperty), SORT);
+  }
+
+  public static ElideParamListMatcher<?> hasIncludes(List<String> includes) {
+    return new ElideParamListMatcher<>(containsInAnyOrder(includes.stream().map(Matchers::equalTo).collect(Collectors.toList())), INCLUDE);
+  }
+
+  public static ElideParamMatcher hasPageSize(int pageSize) {
+    return new ElideParamMatcher(containsString(String.format("page[size]=%d", pageSize)));
+  }
+
+  public static ElideParamMatcher hasPageNumber(int pageNumber) {
+    return new ElideParamMatcher(containsString(String.format("page[number]=%d", pageNumber)));
+  }
+
+  public static ElideParamMatcher hasPageTotals() {
+    return new ElideParamMatcher(containsString("page[totals]"));
+  }
+
+  public static ElideIdMatcher hasId(int id) {
+    return new ElideIdMatcher(matchesPattern(String.format("^/data/\\w+/%d.*", id)));
+  }
+
+  public static ElideIdMatcher hasRelationship(String relationship) {
+    return new ElideIdMatcher(matchesPattern(String.format("^/data/\\w+/\\d+/%s.*", relationship)));
+  }
+
+  @Value
+  public static class ElideParamMatcher implements ArgumentMatcher<ElideNavigatorOnCollection<?>> {
+    Matcher<String> matcher;
+
+    @Override
+    public boolean matches(ElideNavigatorOnCollection<?> argument) {
+      return matcher.matches(argument.build());
+    }
+  }
+
+  @Value
+  public static class ElideIdMatcher implements ArgumentMatcher<ElideNavigatorOnId<?>> {
+    Matcher<String> matcher;
+
+    @Override
+    public boolean matches(ElideNavigatorOnId<?> argument) {
+      return matcher.matches(argument.build());
+    }
+  }
+
+  @Value
+  public static class ElideParamListMatcher<T extends ElideEndpointBuilder<?>> implements ArgumentMatcher<T> {
+    private static final String PARAM_DELIMITER = "&";
+    private static final String PROPERTY_DELIMITER = ",";
+    Matcher<Iterable<? extends String>> matcher;
+    String paramPrefix;
+
+    @Override
+    public boolean matches(T argument) {
+      String endpoint = argument.build();
+      assertThat(endpoint, containsString(paramPrefix));
+      int startIndex = endpoint.indexOf(paramPrefix) + paramPrefix.length();
+      int endIndex = endpoint.indexOf(PARAM_DELIMITER, startIndex);
+      if (endIndex == -1) {
+        endIndex = endpoint.length();
+      }
+      List<String> properties = Arrays.asList(endpoint.substring(startIndex, endIndex).split(PROPERTY_DELIMITER));
+      return matcher.matches(properties);
+    }
+  }
+
+}
```

Comment:
I have to check in my IDE how and where `matcher` is set but it seems fishy, same for `paramPrefix`

Reply:
matcher is defined when in the constructor. It is a hamcrest matcher. The paramPrefix is similar and used to define which query list is examined.

What seems fishy?

Reply:
I didn't realize they are `@Value`.